internal/tests/testrepository/header_testrepository.go

```diff
@@ -193,6 +195,93 @@ func (r *HeaderTestRepository) GetChainBetweenTwoHashes(low string, high string)
 	return headers, nil
 }
 
+// GetMerkleRoots returns ExclusiveStartKey pagination of batchSize size with merkle roots from lastEvaluatedKey which
+// is the last merkleroot of the block that a client has processed
+func (r *HeaderTestRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey string) (*domains.MerkleRootsESKPagedResponse, error) {
+	// Order headers by height ASC
+	sort.Slice(*r.db, func(i, j int) bool {
+		return (*r.db)[i].Height < (*r.db)[j].Height
+	})
+
+	// Check if lastEvaluatedKey is the same as the last element's MerkleRoot
+	if lastEvaluatedKey != "" && (*r.db)[len(*r.db)-1].MerkleRoot.String() == lastEvaluatedKey {
+		// Return empty content since we have reached the end
+		return &domains.MerkleRootsESKPagedResponse{
+			Page: domains.ExclusiveStartKeyPageInfo{
+				TotalElements:    int32(len(*r.db)),
+				Size:             0,
+				LastEvaluatedKey: "",
+			},
+			Content: []domains.MerkleRootsResponse{},
+		}, nil
+	}
+
+	// Find the starting index based on the lastEvaluatedKey
+	startIdx := slices.IndexFunc(*r.db, func(c domains.BlockHeader) bool { return c.MerkleRoot.String() == lastEvaluatedKey })
```

Comment:
Maybe with a inline function in another line?
```suggestion
	startIdx := slices.IndexFunc(*r.db, func(c domains.BlockHeader) bool { 
	    return c.MerkleRoot.String() == lastEvaluatedKey 
	})
```

Reply:
sure thing, moved